src/main/java/io/vertx/ext/asyncsql/impl/AsyncSQLConnectionImpl.java

```diff
@@ -183,90 +298,107 @@ public void close() {
   }
 
   @Override
   public SQLConnection rollback(Handler<AsyncResult<Void>> handler) {
-    return endAndStartTransaction("ROLLBACK", handler);
-  }
-
-  @Override
-  public SQLConnection setQueryTimeout(int i) {
-    throw new UnsupportedOperationException("Not implemented");
+    if (!inAutoCommit.get()) {
+      sendStatementAndFulfill("ROLLBACK", null, handler);
+    } else {
+      handler.handle(Future.failedFuture(new IllegalStateException("Not in transaction currently")));
+    }
+    return this;
   }
 
   @Override
-  public SQLConnection setTransactionIsolation(TransactionIsolation transactionIsolation, Handler<AsyncResult<Void>> handler) {
-    throw new UnsupportedOperationException("Not implemented");
-  }
+  public SQLConnection batch(List<String> sqlStatements, Handler<AsyncResult<List<Integer>>> handler) {
+    if (options != null && options.getQueryTimeout() > 0) {
+      final long timerId = vertx.setTimer(options.getQueryTimeout(), t -> {
+        handler.handle(Future.failedFuture("Query timeout"));
+      });
 
-  @Override
-  public SQLConnection getTransactionIsolation(Handler<AsyncResult<TransactionIsolation>> handler) {
-    throw new UnsupportedOperationException("Not implemented");
-  }
+      batch(0, timerId, new ArrayList<>(sqlStatements.size()), sqlStatements, handler);
+    } else {
+      batch(0, -1, new ArrayList<>(sqlStatements.size()), sqlStatements, handler);
+    }
 
-  @Override
-  public SQLConnection batch(List<String> sqlStatements, Handler<AsyncResult<List<Integer>>> handler) {
-    // This should be simple in postgres, since it is just append the query separator after each query and send as a big
-    // sql statement, however it does not seem to work on mysql
-    throw new UnsupportedOperationException("Not implemented");
+    return this;
   }
 
-  @Override
-  public SQLConnection batchWithParams(String sqlStatement, List<JsonArray> args, Handler<AsyncResult<List<Integer>>> handler) {
-    // This should be simple in postgres, since it is just append the query separator after each query and send as a big
-    // sql statement, however it does not seem to work on mysql
-    throw new UnsupportedOperationException("Not implemented");
-  }
+  private void batch(final int idx, final long timerId, final List<Integer> results, List<String> sqlStatements, Handler<AsyncResult<List<Integer>>> handler) {
+    if (idx == sqlStatements.size()) {
+      // stop condition
+      if (timerId == -1 || vertx.cancelTimer(timerId)) {
+        // cancel the running timer if it is valid or still active
+        handler.handle(Future.succeededFuture(results));
+      }
+      return;
+    }
 
-  @Override
-  public SQLConnection batchCallableWithParams(String sqlStatement, List<JsonArray> inArgs, List<JsonArray> outArgs, Handler<AsyncResult<List<Integer>>> handler) {
-    // No idea how to implement this
-    throw new UnsupportedOperationException("Not implemented");
+    // run the next statement
```

Comment:
True, the issue here is that the underlying driver does not support it like the jdbc API. So the simplest way is to send in a sequence or as you wrote "rewrite" the queries. We could agree to do the rewrite as a second step, since it also requires the same work for postgres